sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java

```diff
@@ -95,17 +94,12 @@ private Bounded(@Nullable String name, BoundedSource<T> source) {
     }
 
     @Override
-    protected Coder<T> getDefaultOutputCoder() {
-      return source.getDefaultOutputCoder();
-    }
-
-    @Override
     public final PCollection<T> expand(PBegin input) {
       source.validate();
 
       return PCollection.<T>createPrimitiveOutputInternal(input.getPipeline(),
           WindowingStrategy.globalDefault(), IsBounded.BOUNDED)
-          .setCoder(getDefaultOutputCoder());
+          .setCoder(source.getDefaultOutputCoder());
```

Comment:
PCollection.setCoder() does not allow passing a null argument, but source.getDefaultOutputCoder() is nullable.